lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -4,9 +4,31 @@ module ActiveModel
   class Serializer
     class Adapter
       class JsonApi < Adapter
+        @root = :data
+        DEFAULT_ATTRIBUTES = [:type, :id]
+
+        def self.params(permitted, associations)
+          relationships = {}
+          associations.each do |assoc|
+             relationships[assoc] = {}
+             relationships[assoc][@root] = DEFAULT_ATTRIBUTES
+           end
+
+          return :type, {attributes: permitted}, {relationships: relationships}
+        end
+
+        def self.parse(params)
+          attrs, assoc = {}, {}
+          attrs = params['attributes'] if params['attributes']
+          assoc = params['relationships'].map {|a| {a.shift => a.first['data']['type'].camelize.safe_constantize.find(a.shift['data']['id'])}} if params['relationships']
```

Comment:
As @GCorbel noticed,
```ruby
assoc = params['relationships'].map {|a| {a.shift => a.first['data']['type'].camelize.safe_constantize.find(a.shift['data']['id'])}} if params['relationships']
```
should be
```ruby
assoc = params['relationships'].map {|a| {a.shift => a.first['data']['type'].camelize.singularize.safe_constantize.find(a.shift['data']['id'])}} if params['relationships']
```
since the JSONAPI `type` attribute can be plural (and it it serialized in a plural form by active_model::serializer), but the corresponding `ActiveModel` class name is singular.